sklearnex/preview/decomposition/pca.py

```diff
@@ -1,372 +0,0 @@
-                        "Only 'full' solver is supported.",
-                    ),
-                    (not issparse(X), "oneDAL PCA does not support sparse inputs"),
-                    (
-                        X.shape[1] / X.shape[0] < 2,
-                        "The shape of X does not satisfy oneDAL requirements: "
-                        "number of features / number of samples >= 2",
-                    ),
-                ]
-            )
-            return patching_status
-        elif method_name == "transform":
-            patching_status = PatchingConditionsChain(
-                f"sklearn.decomposition.{class_name}.{method_name}"
-            )
-            patching_status.and_conditions(
-                [
-                    (hasattr(self, "_onedal_estimator"), "oneDAL model was not trained"),
-                ]
-            )
-            return patching_status
-        raise RuntimeError(f"Unknown method {method_name} in {self.__class__.__name__}")
-
-    def _onedal_cpu_supported(self, method_name, *data):
-        return self._onedal_supported(method_name, *data)
-
-    def _onedal_gpu_supported(self, method_name, *data):
-        return self._onedal_supported(method_name, *data)
-
-    def _onedal_fit(self, X, y=None, queue=None):
-        if self.n_components == "mle" or self.n_components is None:
-            onedal_n_components = min(X.shape)
-        elif 0 < self.n_components < 1:
-            onedal_n_components = min(X.shape)
-        else:
-            onedal_n_components = self.n_components
-
-        onedal_params = {
-            "n_components": onedal_n_components,
-            "is_deterministic": True,
-            "method": "precomputed",
-        }
-        self._onedal_estimator = onedal_PCA(**onedal_params)
-        self._onedal_estimator.fit(X, queue=queue)
-        self._save_attributes()
-
-        U = None
-        S = self.singular_values_
-        V = self.components_
-
-        return U, S, V
-
-    def _onedal_predict(self, X, queue=None):
-        return self._onedal_estimator.predict(X, queue)
-
-    def _onedal_transform(self, X):
-        X = _check_array(X, dtype=[np.float64, np.float32], ensure_2d=True, copy=False)
-
-        if hasattr(self, "n_features_in_"):
-            if self.n_features_in_ != X.shape[1]:
-                raise ValueError(
-                    f"X has {X.shape[1]} features, "
-                    f"but {self.__class__.__name__} is expecting "
-                    f"{self.n_features_in_} features as input"
-                )
-        elif hasattr(self, "n_features_"):
-            if self.n_features_ != X.shape[1]:
-                raise ValueError(
-                    f"X has {X.shape[1]} features, "
-                    f"but {self.__class__.__name__} is expecting "
-                    f"{self.n_features_} features as input"
-                )
-
-        # Mean center
-        X_centered = X - self.mean_

```

Comment:
is dropping any of this any issue? or is it covered somewhere

Reply:
mean centering is covered in oneDAL and rest is just relocated and refactored